platforms/defaults_windows_test.go

```diff
@@ -109,27 +119,107 @@ func TestMatchComparerMatch_WCOW(t *testing.T) {
 			platform: imagespec.Platform{
 				Architecture: "amd64",
 				OS:           "windows",
-				// Use an nonexistent Windows build so we don't get a match. Ws2019's build is 17763/
-				OSVersion: "10.0.17762.1",
+				// Try a build number smaller than the current one:
+				OSVersion: fmt.Sprintf("10.0.%d.1", build-1),
 			},
 			match: false,
 		},
 		{
 			platform: imagespec.Platform{
 				Architecture: "amd64",
 				OS:           "windows",
-				// Use an nonexistent Windows build so we don't get a match. Ws2019's build is 17763/
-				OSVersion: "10.0.17764.1",
-			},
+				// Try a build number larger than the current one:
+				OSVersion: fmt.Sprintf("10.0.%d.1", build+1)},
 			match: false,
 		},
 		{
 			platform: imagespec.Platform{
 				Architecture: "amd64",
 				OS:           "windows",
 			},
+			// If there is no platform.OSVersion, we assume it can run:
```

Comment:
Fair point.. are there many Windows images without this? What did docker do here?

Reply:
The current behavior in upstream Docker is to accept images with empty or misformatted OSVersion strings. (i.e. [if `strings.Split(OSVersion, ".")` yields less than 3 components](https://github.com/moby/moby/blob/master/distribution/pull_v2_windows.go#L129-L139))

[moby/moby#44489](https://github.com/moby/moby/pull/44489) only [adds a log message](https://github.com/moby/moby/pull/44489/files#diff-6019c68f8c9392b97b32ce83ecaa765fdd517946fc9685ba523182f647cd2981R146-R147).